### ISSUE TITLE
How to deal with data that is not in the dataset in ML.NET machine learning model?

### ISSUE THREAD

**Issue by nishaesubhash** [dotnet/machinelearning#4137]
I had created an answer grade prediction model in ML.NET(tried different regression algorithms and multi class classification algorithm). I have ID, question, answer and grade in my dataset. Inputs to the model are question and answer. Model predicts the grade for the answer. If we provide an input(answer) that is in the dataset, then the model predicts the grade accurately. But when the answer is completely incorrect and not matching with any of the answers in the dataset, then the model is not predicting the grade accurately.

Any ideas on how to deal with this case?

**Comment by SnakyBeaky** [comment]
Not sure I fully understand what you're explaining, but it sounds like your model is overfitting on your training data.

You're getting great accuracy with samples that your model has seen and learned from, but once you provide a sample that was not in the training set, your model fails to generalize and predict.

There's many reasons this could happen and many ways to avoid it. I would suggest taking a look at regularization and collecting a bigger data set for training (and augmenting it if possible).

**Comment by codemzs** [comment]
Thanks @SnakyBeaky this does seem like an overfiting problem. It could you may need a more data so that your model is generalized.
